lib/jnpr/junos/utils/sw.py

```diff
@@ -967,11 +968,14 @@ def reboot(self, in_min=0, at=None, all_re=True):
 
         .. todo:: need to better handle the exception event.
         """
+        if self._is_linux:
```

Comment:
This line can be changed from `if self._is_linux` to `dev.facts['_is_linux']`. Because `_is_linux` is defined to be a known fact (at `lib/jnpr/junos/facts/is_linux.py` line 9), the expression `'_is_linux' in dev.facts` will always return `True`. That's because the facts system does "on demand" gathering of facts.